tests/core/policies/test_rule_policy.py

```diff
@@ -642,49 +678,49 @@ def test_contradicting_rules_and_stories():
     )
 
     with pytest.raises(InvalidRule) as execinfo:
-        policy.train([GREET_RULE, anti_greet_story], domain, RegexInterpreter())
+        policy.train([GREET_RULE, anti_greet_story], domain)
 
     assert all(
         name in execinfo.value.message
         for name in {utter_anti_greet_action, anti_greet_story.sender_id}
     )
 
 
-def test_rule_policy_has_max_history_none():
-    policy = RulePolicy()
+def test_rule_policy_has_max_history_none(policy: RulePolicy):
     assert policy.featurizer.max_history is None
 
 
-def test_all_policy_attributes_are_persisted(tmpdir: Path):
-    priority = 5
-    lookup = {"a": "b"}
-    core_fallback_threshold = 0.451231
-    core_fallback_action_name = "action_some_fallback"
-    enable_fallback_prediction = False
+def test_all_policy_attributes_are_persisted(
+    trained_rule_policy: RulePolicy,
+    default_model_storage: ModelStorage,
+    default_execution_context: ExecutionContext,
+    resource: Resource,
+):
+    lookup = trained_rule_policy.lookup
 
-    policy = RulePolicy(
-        priority=priority,
-        lookup=lookup,
-        core_fallback_threshold=core_fallback_threshold,
-        core_fallback_action_name=core_fallback_action_name,
-        enable_fallback_prediction=enable_fallback_prediction,
+    loaded = RulePolicy.load(
+        RulePolicy.get_default_config(),
```

Comment:
Btw, It's a bit weird that we don't store the old config. One could trainand save something with e.g. embeddingdim=100 and during load decide to set that with embeddingdim=200. 🤔

Reply:
This is stored as part of the graph schema. The individual graph components don't have to store it themselves